### PR TITLE
Make use of compstate validation action

### LINKED ISSUE #12: Create a GitHub Action to validate compstates
Compstates contain scoring code, which is pretty important. However due to the compstates being short-lived one-off repos it doesn't feel worth it to add CI to them.
If however we had a GitHub Action (or other CI template, though I think an Action could make it simplest) that could be trivially included to get useful checking in CI this could provide more confidence in the scoring code.

### PR DESCRIPTION
**Pull request by PeterJCLaw**
When reviewing please also review the action which is used -- https://github.com/PeterJCLaw/srcomp-validate-action

Fixes https://github.com/PeterJCLaw/srcomp/issues/12

Strongly encourage reviewing only the diff! I'm leaving the history here as potentially useful history if we need to develop the action more, but will squash for merge.

### CODE REVIEW

File: .github/workflows/validate.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  validate:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
```
Comment (sedders123) [inline .github/workflows/validate.yml]: May as well start on latest version
```suggestion
        uses: actions/checkout@v3
```

**Review by sedders123 — APPROVED** [review]
Both this PR and the action look good!

On the actions front I did wonder about whether we should be including the Setup Python action within it as this feels like something consumers may want to specify themselves (for example to set a specific Python version). But it is nice for our compstate repos to not to have to worry about that. I ended up that it was probably fine and we can look at tweaking only if it actually starts causing issues.

File: .github/workflows/validate.yml (hunk 1 above)
Comment (PeterJCLaw) [reply .github/workflows/validate.yml]: ah yes.

**Comment by PeterJCLaw** [comment]
> On the actions front I did wonder about whether we should be including the Setup Python action within it as this feels like something consumers may want to specify themselves (for example to set a specific Python version). But it is nice for our compstate repos to not to have to worry about that. I ended up that it was probably fine and we can look at tweaking only if it actually starts causing issues.

Yeah, I think there's a lot here which could be parameterised if we wanted:
- python version
- path to the tests
- _maybe_ additional test dependencies, though I'm not sure I want to encourage that
- whether to run the tests first or the validation first
- whether to run the validation at all or just check the compstate loads; there's a general thing here around more control over the error level too -- the validation already has warnings (which allow a zero exit code) and errors though no way to control them

For now I was going for the simplest useful increment, which I think this is.

**Comment by PeterJCLaw** [comment]
Squashed as https://github.com/srobo/sr2022-comp/commit/2200dc03dc4b3aa3d9993477c5aadedd33b44c1f, though I've just spotted that I'd assumed that `master` was the main branch here, but it's `main`.

**Pull request closed without merging**